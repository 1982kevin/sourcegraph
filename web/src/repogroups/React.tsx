import { RepogroupMetadata } from './types'
import { SearchPatternType } from '../../../shared/src/graphql/schema'
import * as React from 'react'
export const react: RepogroupMetadata = {
    title: 'React',
    name: 'react',
    url: '/react',
    description:
        "Explore React repositories on GitHub. The search examples show how to use React Hook useState with various data types.",
    examples: [
        {
            title: 'Find imports of useState with regex search',
            exampleQuery: <>import [^;]+useState[^;]+ from 'react'</>,
            rawQuery: "import [^;]+useState[^;]+ from 'react'",
            patternType: SearchPatternType.regexp,
        },
        {
            title: 'Examples of useState with an object as the input parameter',
            description: `useState takes a single argument and can accept a primitive, an array or an object.
            Syntax for usage is [state_value, function_to_update_state_value] = useState (initial_state_value).`,
            exampleQuery: <>{'useState({:[string]})'}</>,
            rawQuery: 'useState({:[string]}) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter',
            exampleQuery: <>useState(:[string])</>,
            rawQuery: 'useState(:[string]) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter in only TypeScript files',
            exampleQuery: (
                <>
                    useState(:[string])
                    <span className="search-keyword">lang:</span>typescript
                </>
            ),
            rawQuery: 'useState(:[string]) count:1000 lang:typescript',
            patternType: SearchPatternType.structural,
        },
    ],
    homepageDescription: 'Learn React with code examples.',
    homepageIcon: 'https://cdn4.iconfinder.com/data/icons/logos-3/600/React.js_logo-512.png',
}
