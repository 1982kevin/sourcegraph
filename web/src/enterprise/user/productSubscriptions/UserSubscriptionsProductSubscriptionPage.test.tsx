import React from 'react'
import * as GQL from '../../../../../shared/src/graphql/schema'
import { UserSubscriptionsProductSubscriptionPage } from './UserSubscriptionsProductSubscriptionPage'
import { of } from 'rxjs'
import { MemoryRouter } from 'react-router'
import { createMemoryHistory } from 'history'
import { mount } from 'enzyme'

jest.mock('./BackToAllSubscriptionsLink', () => ({
    BackToAllSubscriptionsLink: 'BackToAllSubscriptionsLink',
}))

jest.mock('./UserProductSubscriptionStatus', () => ({
    UserProductSubscriptionStatus: 'UserProductSubscriptionStatus',
}))

jest.mock('./ProductSubscriptionBilling', () => ({
    ProductSubscriptionBilling: 'ProductSubscriptionBilling',
}))

jest.mock('./ProductSubscriptionHistory', () => ({
    ProductSubscriptionHistory: 'ProductSubscriptionHistory',
}))

jest.mock('../../../tracking/eventLogger', () => ({
    eventLogger: { logViewEvent: () => undefined },
}))

describe('UserSubscriptionsProductSubscriptionPage', () => {
    test('renders', () => {
        expect(
            mount(
                <MemoryRouter>
                    <UserSubscriptionsProductSubscriptionPage
                        user={{ settingsURL: '/u' }}
                        match={{ isExact: true, params: { subscriptionUUID: 's' }, path: '/p', url: '/p' }}
                        _queryProductSubscription={() =>
                            of<GQL.IProductSubscription>({
                                __typename: 'ProductSubscription',
                            } as GQL.IProductSubscription)
                        }
                        history={createMemoryHistory()}
                    />
                </MemoryRouter>
            )
        ).toMatchSnapshot()
    })
})
