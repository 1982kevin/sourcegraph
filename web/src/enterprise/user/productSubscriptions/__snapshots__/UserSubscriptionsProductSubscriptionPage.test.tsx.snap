// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSubscriptionsProductSubscriptionPage renders 1`] = `
<MemoryRouter>
  <Router
    history="[History]"
  >
    <UserSubscriptionsProductSubscriptionPage
      _queryProductSubscription={[Function]}
      history="[History]"
      match={
        Object {
          "isExact": true,
          "params": Object {
            "subscriptionUUID": "s",
          },
          "path": "/p",
          "url": "/p",
        }
      }
      user={
        Object {
          "settingsURL": "/u",
        }
      }
    >
      <div
        className="user-subscriptions-product-subscription-page"
      >
        <PageTitle
          title="Subscription"
        />
        <div
          className="d-flex align-items-center justify-content-between"
        >
          <BackToAllSubscriptionsLink
            user={
              Object {
                "settingsURL": "/u",
              }
            }
          />
        </div>
        <h2>
          Subscription 
        </h2>
        <div
          className="card mt-3"
        >
          <div
            className="card-header"
          >
            Billing
          </div>
          <div
            className="card-body"
          >
            <span
              className="text-muted "
            >
              No billing information is associated with this subscription.
               
              <a
                href="mailto:sales@sourcegraph.com?subject=Billing%20for%20subscription%20undefined"
              >
                Contact sales
              </a>
               
              for help.
            </span>
          </div>
        </div>
        <div
          className="card mt-3"
        >
          <div
            className="card-header"
          >
            History
          </div>
          <ProductSubscriptionHistory
            productSubscription={
              Object {
                "__typename": "ProductSubscription",
              }
            }
          />
        </div>
      </div>
    </UserSubscriptionsProductSubscriptionPage>
  </Router>
</MemoryRouter>
`;
